pkg/crd/spec.go

```diff
@@ -36,6 +39,102 @@ type SpecMarker interface {
 	ApplyToCRD(crd *apiext.CustomResourceDefinitionSpec, version string) error
 }
 
+type VersionType string
+
+const (
+	GA    VersionType = "ga"
+	Beta  VersionType = "beta"
+	Alpha VersionType = "alpha"
+)
+
+type VersionName string
+
+// Value creates a float64 type of value to represent a VersionName
+func (v VersionName) Value() float64 {
+	// remove the first character, `v`
+	name := string(v)[1:]
+	name = strings.Replace(name, "beta", ".", -1)
+	name = strings.Replace(name, "alpha", ".", -1)
+	result, _ := strconv.ParseFloat(name, 64)
```

Comment:
eeek.  versions are not floats, and treating them as such is a bad practice.  For example: beta1 is not the same as beta10, but in float terms they would be.  You'll need to find a better way to do this.

Reply:
removed